Framework/MDAlgorithms/src/ConvertToDiffractionMDWorkspace3.cpp

```diff
@@ -114,6 +114,18 @@ void ConvertToDiffractionMDWorkspace3::calculateExtentsFromData(
   std::replace(maxVal.begin(), maxVal.end(), INF, DEFAULT_BOUND);
   std::replace(minVal.begin(), minVal.end(), MAX_DBL, -DEFAULT_BOUND);
   std::replace(maxVal.begin(), maxVal.end(), -MAX_DBL, DEFAULT_BOUND);
+
+  // Increases bounds by a small margin and ensures they aren't too close to
+  // zero. This is to prevent events from being incorrectly discarded as out
+  // of bounds by calcMatrixCoord functions.
```

Comment:
You pointed out a related bug with index calculation. Comment should include link/note to the `MDBox` box index calculation code error that is coupled with events on upper boundaries.